internal/schemautil/userconfig/userconfig_test.go

```diff
@@ -14,6 +16,167 @@ import (
 )
 
+const (
+	typePrimitive      = "primitive"
+	typeObject         = "obj"
+	typeArray          = "array"
+	typeArrayOfObjects = "array-obj"
+	typeOneOf          = "one-of"
+)
+
+func listOfCorrectTypeResolutions() []string {
+	return []string{
+		typePrimitive,
+		typeObject,
+		typeArray,
+		typeArrayOfObjects,
+		typeOneOf,
+	}
+}
+
+func listOfPrimitiveTypes() []string {
+	return []string{"boolean", "integer", "string", "number"}
+}
+
+func getElementType(m map[string]interface{}) string {
+	t, ok := m["type"]
+	if !ok {
+		return "type-is-empty"
+	}
+
+	switch reflect.TypeOf(t).Kind() {
+	case reflect.Slice, reflect.Array:
+		buf := ""
+		for _, tt := range t.([]interface{}) {
+			tt := tt.(string)
+			if tt == "null" {
+				continue
+			}
+
+			if buf != "" {
+				return typeOneOf
+			}
+
+			if slices.Contains(listOfPrimitiveTypes(), tt) {
+				buf = typePrimitive
+			}
+
+			if tt == "array" {
+				buf = typeArray
+
+				if i, ok := m["items"]; ok {
+					if getElementType(i.(map[string]interface{})) == typeObject {
+						buf = typeArrayOfObjects
+					}
+				}
+			}
+
+			if tt == "object" {
+				buf = typeObject
+			}
+		}
+		return buf
+
+	case reflect.String:
+		tt := t.(string)
+		if slices.Contains(listOfPrimitiveTypes(), tt) {
+			return typePrimitive
+		}
+
+		if tt == "object" {
+			return typeObject
+		}
+
+		if tt == "array" {
+			if i, ok := m["items"]; ok {
+				if getElementType(i.(map[string]interface{})) == typeObject {
+					return typeArrayOfObjects
+				}
+			}
+			return typeArray
+		}
+	default:
+		return "incorrect-element-type"
+	}
+
+	return "type-undefined"
+}
+
+func generateToApi(n string, m map[string]interface{}) error {
+	f := jen.NewFile("toapi")
+	f.HeaderComment("Code generated by internal/schemautil/userconfig/userconfig_test.go; DO NOT EDIT.")
+
+	smk := maps.Keys(m)
+	slices.Sort(smk)
+
+	for _, k := range smk {
+		kp := strcase.ToGoPascal(k)
+		fn := fmt.Sprintf("%s%sToAPIBlueprint", n, kp)
+		f.Commentf("%s is a generated function converts TF schema to API format %s %s.", fn, k, n)
+
+		va, ok := m[k].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		pa, ok := va["properties"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		f.
+			Func().
+			Id(fmt.Sprintf("%s%sToAPIBlueprint", n, kp)).Params().
+			Id("map[string]interface{}").
+			Block(
+				jen.Return(
+					jen.Map(jen.String()).Interface().Values(generateToApiElements(pa)),
+				),
+			).
+			Line()
+	}
+
+	if err := f.Save(fmt.Sprintf("toapi/%s.go", strcase.ToSnake(n)+"s_to_api")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func generateToApiElements(m map[string]interface{}) jen.Dict {
+	d := make(jen.Dict)
+	for k, v := range m {
+		if _, ok := v.(map[string]interface{}); !ok {
+			continue
+		}
+
+		va, ok := m[k].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		switch getElementType(va) {
+		case typePrimitive, typeArray, typeOneOf:
+			d[jen.Lit(k)] = jen.Lit("primitive")
```

Comment:
this has to be replaced with `struct{}`: https://medium.com/@l.peppoloni/how-to-improve-your-go-code-with-empty-structs-3bd0c66bc531

or do we plan to have anything else instead of "primitive" over here?

Reply:
nope, only this

Reply:
please, change this then 🙂